azkaban-common/src/main/java/azkaban/executor/ExecutorLoader.java

```diff
@@ -47,6 +48,82 @@ public ExecutableFlow fetchExecutableFlow(int execId)
       String flowContains, String userNameContains, int status, long startData,
       long endData, int skip, int num) throws ExecutorManagerException;
 
+  /**
+   * Fetch all executors from executors table with active = true
+   *
+   * @return List<Executor>
+   * @throws ExecutorManagerException
+   */
+  public List<Executor> fetchActiveExecutors() throws ExecutorManagerException;
+
+  /**
+   * Fetch executor from executors with a given (host, port)
+   *
+   * @return Executor
+   * @throws ExecutorManagerException
+   */
+  public Executor fetchExecutor(String host, int port)
+    throws ExecutorManagerException;
+
+  /**
+   * Fetch executor from executors with a given executorId
+   *
+   * @return Executor
+   * @throws ExecutorManagerException
+   */
+  public Executor fetchExecutor(int executorId) throws ExecutorManagerException;
+
+  /**
+   * create an executor and insert in executors table. Fails, if a executor with
+   * (host, port) already exist
+   *
+   * @return Executor
+   * @throws ExecutorManagerException
+   */
+  public Executor addExecutor(String host, int port)
+    throws ExecutorManagerException;
+
+  /**
+   * inactivate an executor in executors table
+   *
+   * @param executorId
+   * @throws ExecutorManagerException
+   */
+  public void inactivateExecutor(int executorId)
+    throws ExecutorManagerException;
+
+  /**
+   * Re-activate an executor in executors table
+   *
+   * @param executorId
+   * @throws ExecutorManagerException
+   */
+  void activateExecutor(int executorId) throws ExecutorManagerException;
+
+  /**
+   * Log an event in executor_event audit table
+   *
+   * @param executor
+   * @param type
+   * @param user
+   * @param message
+   * @return isSuccess
+   */
+  boolean postEvent(Executor executor, EventType type, String user,
+    String message);
+
+  /**
+   * Fetch num events associated with a given executor, starting from skip
+   *
+   * @param executor
+   * @param num
+   * @param skip
+   * @return List<ExecutorLogEvent>
+   * @throws ExecutorManagerException
+   */
+  List<ExecutorLogEvent> getExecutorEvents(Executor executor, int num, int skip)
```

Comment:
Does this method return null or empty list when there are not matches?  Please add documentation about this behavior.